### DISCUSSION TITLE
How to execute multiple tasks in parallel with taskflow api

### DISCUSSION THREAD

**Discussion by aamster** [apache/airflow#28906] [Q&A] [answered]
With taskflow API it is unclear how to execute tasks in parallel..

The example 

order_data = extract()
    order_summary = transform(order_data)
    load(order_summary["total_order_value"])

is linear. Is it possible to write parallel task execute like in old api

A >> [b, c]

**Comment by potiuk** [comment]
Task flow is a not a new way. Tasks are always executed in parallel (as much as resources and other criteria - such as pools and queus and parallelism allow) unless there is a dependency between them. nothing changed here. `A >> [b,c]` does not mean that  [b,c] tasks are executed in parallel. It means that both b and c depend on A executed before. that's it.

**Reply by potiuk** [reply]
And again - if you think it's not described well - feel free to make PR clarifying there where you understood it wrongly. PRs to improve docs from users like you (especially when they got help and their questions were answered) are always welcome.

**Comment by ferruzzi** [comment, marked as answer]
The TaskFlow API will arrange them in parallel if it's possible.  In that case it is parallel because the requirements are organized that way.  You must complete those tasks in that order.  If the tasks don't have to be sequential then TaskFlow is smart enough to know that.  See for example:

![image](https://user-images.githubusercontent.com/1920178/212201279-5886e9bd-c06b-4d48-bb58-f908b9b3d56f.png)

results in 

![image](https://user-images.githubusercontent.com/1920178/212201319-58ab6946-6a97-4a1b-9c89-020c4d923c4c.png)

**Reply by ferruzzi** [reply]
Similarly, 
![image](https://user-images.githubusercontent.com/1920178/212201744-fa611659-196e-4891-9ab7-481b061efcdb.png)

results in 
![image](https://user-images.githubusercontent.com/1920178/212201766-2fdc26d6-9d42-41d9-b3c1-e97ea3d9379f.png)
